src/uct/ib/mlx5/ib_mlx5.h

```diff
@@ -143,4 +153,193 @@ static inline struct mlx5_cqe64* uct_ib_mlx5_get_cqe(uct_ib_mlx5_cq_t *cq,
 }
 
 
+/* send WQ */
+typedef struct uct_ib_mlx5_txwq {
+    uint16_t       sw_pi;      /* PI for next WQE */
+    uint16_t       prev_sw_pi; /* PI where last WQE *started*  */
+    unsigned       bf_size;
+    void           *bf_reg;
+    uint32_t       *dbrec;
+    void           *seg;
+    void           *qstart;
+    void           *qend;
+} uct_ib_mlx5_txwq_t;
+
+typedef uint16_t uct_ib_mlx5_index_t;
```

Comment:
if you're adding this, need to use it instead of uint16_t in all places, including rc code. maybe better just to use uint16_t instead.